packages/azure-kusto-ingest/src/resourceManager.ts

```diff
@@ -98,15 +115,27 @@ export class ResourceManager {
 
     async refreshAuthorizationContext(): Promise<string> {
         const now = Date.now();
+        let error: Error | null = null;
         if (!this.authorizationContext?.trim() || !this.authorizationContextLastUpdate || this.authorizationContextLastUpdate + this.refreshPeriod <= now) {
             this.authorizationContext = await this.getAuthorizationContextFromService();
```

Comment:
Can this not similarly fail?

Reply:
And if you do it 3 times might be worth it to refactor into a method

Reply:
it doesnt - see the test ... 
its 2 times but ok